pages/api/roles/index.ts

```diff
@@ -68,12 +69,35 @@ function handler(req: NextApiRequest, res: NextApiResponse) {
   }
 }
 
-// TODO: Optional Search query. Will be done in another PR to unblock others.
-
 async function handleGet(req: NextApiRequest, res: NextApiResponse<ApiResponse<RoleListData[]>>) {
+  const { companyId, searchQuery } = req.query;
+
+  const queryCompanyId = isInteger(companyId as string) ? Number(companyId) : undefined;
+
+  const searchWords = typeof searchQuery === 'string' ? (searchQuery as string).split(' ') : [];
```

Comment:
This looks good to me. But I would prefer `QueryParams` interface to be in the `/types` instead of in the `/pages/api/{handler file}`. There already is `{api name}{api method}Data` (e.g. `RolePostData`) interface in the `/types/roles` path. If we put the `{api name}{api method}QueryParams` (e.g. `RolePostQueryParams) in the `/types/roles` path it will be useful for future client side API work.

Reply:
Wait actually how do you know `searchQuery` is a string?

Reply:
And also since companyId is already casted as a number, `isInteger(companyId)` won't compile I think...

```suggestion
function validateQueryParams(req.query) {
  const { companyId, searchQuery } = req.query;
  if (!isInteger(companyId as string)) return undefined;
  
  return { 
    companyId: Number(companyId), 
    searchQuery: typeof searchQuery === 'string' ? searchQuery : "";
  }
}

async function handleGet(req: NextApiRequest, res: NextApiResponse<ApiResponse<RoleListData[]>>) {
  const queryParams = validateQueryParams(req.query);
  if (!queryParams) { .. reject 400 }

  const { companyId, searchQuery }  = queryParams;
  const searchWords = typeof searchQuery === 'string' ? (searchQuery as string).split(' ') : [];
```
Hmm idk if this is any better...

Reply:
```
type QueryParams = {
  companyId: number;
  searchQuery: string;
}
```
@bharathcs This does not work, and if you try to do this Typescript will complain that these types are not overlapping, since the type defined for `query` is 
```
    query: Partial<{
        [key: string]: string | string[];
    }>;
```
and `number` and `string` do not overlap.

> Wait actually how do you know searchQuery is a string?

We don't, although we can implicitly assume that it will be since we are the ones specifying the query parameters from the frontend. But to be safe we can keep the type-checking `typeof searchQuery === 'string'`. (And I just realised that with these explicit typechecks we don't need the type assertions anymore which means my original comment is irrelevant, but I guess we can still keep `RoleQueryParams`.)